src/org/thoughtcrime/securesms/connect/AccountManager.java

```diff
@@ -115,144 +50,77 @@ public static AccountManager getInstance() {
-        // this function is not needed on the very first account creation.
-        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
-        String prevDbName = sharedPreferences.getString("curr_account_db_name", DEFAULT_DB_NAME);
-        String inCreationDbName = getUniqueDbName(context).getName();
-
-        sharedPreferences.edit().putString("prev_account_db_name", prevDbName).apply();
-        sharedPreferences.edit().putString("curr_account_db_name", inCreationDbName).apply();
-
+        DcHelper.getAccounts(context).addAccount();
         resetDcContext(context);
     }
 
     public boolean canRollbackAccountCreation(Context context) {
-        String prevDbName = PreferenceManager.getDefaultSharedPreferences(context)
-                .getString("prev_account_db_name", "");
-        return !prevDbName.isEmpty();
+        return DcHelper.getAccounts(context).getAll().length > 1;
     }
 
     public void rollbackAccountCreation(Activity activity) {
-        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(activity);
-        String prevDbName = sharedPreferences.getString("prev_account_db_name", "");
-        String inCreationDbName = sharedPreferences.getString("curr_account_db_name", "");
-
-        Account prevAccount = maybeGetAccount(new File(activity.getFilesDir(), prevDbName));
-        if (prevAccount!=null) {
-            new SwitchAccountAsyncTask(activity, R.string.switching_account, prevAccount, inCreationDbName, null).execute();
-        }
-    }
-
-
-    // delete account
-
-    private void deleteAccount(Context context, String dbName) {
-        try {
-            File blobdir = new File(context.getFilesDir(), dbName+"-blobs");
-            String [] blobfiles = blobdir.list();
-            for (String blobfile: blobfiles) {
-                new File(blobdir, blobfile).delete();
-            }
-            blobdir.delete();
+        DcAccounts accounts = DcHelper.getAccounts(activity);
 
-            File dbFile = new File(context.getFilesDir(), dbName);
-            dbFile.delete();
-        } catch(Exception e) {
-            e.printStackTrace();
+        DcContext selectedAccount = accounts.getSelectedAccount();
+        if (selectedAccount.isConfigured() == 0) {
+          accounts.removeAccount(selectedAccount.getAccountId());
         }
 
-        File test = new File(context.getFilesDir(), dbName);
-        if (test.exists()) {
-            String err = String.format("Cannot delete account %s", test.getAbsolutePath());
-            Log.w(TAG, err);
-            Toast.makeText(context, err, Toast.LENGTH_LONG).show();
-        }
+        new SwitchAccountAsyncTask(activity, R.string.switching_account, accounts.getSelectedAccount().getAccountId(), null).execute();
     }
 
 
     // helper class for switching accounts gracefully
 
     private static class SwitchAccountAsyncTask extends ProgressDialogAsyncTask<Void, Void, Void> {
```

Comment:
the `SwitchAccountAsyncTask` might be superfluous now as account switching should be much faster, but i leave that in for now until we actually know that it is _always_ fast.